test/storage/memory.test.ts

```diff
@@ -38,6 +38,30 @@ describe('tests memory storage', () => {
     expect(result2.data?.data).toBe('data');
   });
 
+  // Expects that a when value saved using storage.set() is has his inner properties updated,
+  // a request to storage.get() should return unmodified value.
+  //
+  // https://github.com/arthurfiorette/axios-cache-interceptor/issues/580
+  it('not allow changes by value reference before', async () => {
+    const storage = buildMemoryStorage(false, false, false, true);
+
+    const data = { ...EMPTY_RESPONSE, data: 'data' };
+    await storage.set('key', {
+      state: 'cached',
+      createdAt: Date.now(),
+      ttl: 1000 * 60 * 5, // 5 Minutes
+      data: data
+    });
+
+    data.data = 'another data';
+
```

Comment:
This check below checks if the `find()` clone is working. Please add expect statements with direct access to `storage.data`.